lib/apiservers/portlayer/restapi/handlers/misc_handlers.go

```diff
@@ -16,18 +16,45 @@ package handlers
 
 import (
 	"github.com/go-swagger/go-swagger/httpkit/middleware"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
 	"github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations"
 	"github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/misc"
+	"github.com/vmware/vic/lib/portlayer/exec"
 )
 
 type MiscHandlersImpl struct{}
 
 // Configure assigns functions to all the miscellaneous api handlers
 func (handler *MiscHandlersImpl) Configure(api *operations.PortLayerAPI, handlerCtx *HandlerContext) {
 	api.MiscPingHandler = misc.PingHandlerFunc(handler.Ping)
+	api.MiscGetVCHInfoHandler = misc.GetVCHInfoHandlerFunc(handler.GetVCHInfo)
 }
 
 // Ping sends an OK response to let the client know the server is up
 func (handler *MiscHandlersImpl) Ping() middleware.Responder {
 	return misc.NewPingOK().WithPayload("OK")
 }
+
+func (handler *MiscHandlersImpl) GetVCHInfo() middleware.Responder {
+	n := new(int64)
+	m := new(int64)
+	os := new(string)
+	osVer := new(string)
+	hostProduct := new(string)
+
+	*n = exec.VCHConfig.VCHMhz
+	*m = exec.VCHConfig.VCHMemoryLimit
+	*os = exec.VCHConfig.HostOS
+	*osVer = exec.VCHConfig.HostOSVersion
+	*hostProduct = exec.VCHConfig.HostProductName
+
+	vchInfo := &models.VCHInfo{
```

Comment:
why we are not using exec.VCHConfig values, am i missing something (eg; CPUMhz: &exec.VCHConfig.VCHMhz)?